siff-dr/siff-header.go

```diff
@@ -35,28 +35,49 @@ func setSiffFields(packet *netfilter.NFPacket, flags layers.IPv4Flag, capabiliti
 	}
 
 	/* Modify the ip layer information */
-	// compute new IHL and length
-	var IHLchange uint8 = (*ipLayer).IHL
+	var IHLchange uint16 = uint16((*ipLayer).IHL)
 
+	// compute new IHL and length
 	if (flags & 0x03) == IS_SIFF {
-		(*ipLayer).IHL = 7
-	} else if (flags & 0x03) == (IS_SIFF | CAPABILITY_UPDATE) {
 		(*ipLayer).IHL = 9
+	} else if (flags & 0x03) == (IS_SIFF | CAPABILITY_UPDATE) {
+		(*ipLayer).IHL = 11
+	} else {	// evil
+		(*ipLayer).IHL = 7
 	}
 
-	/* change the total length by the change in IHL * 4 to convert from
-	   32-bit words to bytes */
-	IHLchange = (*ipLayer).IHL - IHLchange
+	IHLchange = uint16((*ipLayer).IHL) - IHLchange
 	if IHLchange != 0 {
-		(*ipLayer).Length = uint16((*ipLayer).IHL) * 4
+		(*ipLayer).Length = (*ipLayer).Length + IHLchange * 4
 	}
 
-	// set the flags, preserving the first flag bit in case it is used
-	(*ipLayer).IHL = flags
+	/* change the total length by the change in IHL * 4 to convert from
+	   32-bit words to bytes */
+	(*ipLayer).Length = uint16((*ipLayer).IHL) * 4
+
+	// set the flags option
+	var flagOption layers.IPv4Option
+	flagOption.OptionType = 86
+	flagOption.OptionLength = 8
+	var flag_array [6]byte = [6]byte{0, 0, 0, 0, 0, 0}
+	if (flags & EVIL) == EVIL {
+		flag_array[0] = 1
+	}
+	if (flags & EXP) == EXP {
+		flag_array[1] = 1
+	}
+	if (flags & IS_SIFF) == IS_SIFF {
+		flag_array[2] = 1
```

Comment:
Which flag do you mean? The comment doesn't specify which line number it is?

As far as options type, this link (http://www.tcpipguide.com/free/t_IPDatagramOptionsandOptionFormat.htm) seems to imply that the options type is used somewhere by something, so I'm hesitant to touch it beyond what we know works in case it breaks anything, given the short deadline.